### PR TITLE
Fix for hard-coded service account name in rolebindings

### PR DESCRIPTION
**Pull request by BlaD200**
**What this PR does / why we need it**:
There is a problem with the service account name usage in templates. It's not possible to change the name from the default "seldon-manager" to something else, because in deployment phase this error occurs:
`"Error creating: pods "seldon-controller-manager-xxx-" is forbidden: error looking up service account xxx/seldon-manager: serviceaccount "seldon-manager" not found".`
So the fix is to use the name from the `serviceAccount.name` variable defined in values.yaml instead of the hard-coded one.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @BlaD200. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by agrski — COMMENTED** [review]
I've confirmed these are the only places which use the hard-coded value directly, and that the path to the value is accurate.
```
$ rg '\bseldon-manager\b(\s|$)' helm-charts/ 
helm-charts/seldon-core-operator/values.yaml
102:  name: seldon-manager

helm-charts/seldon-core-operator/templates/clusterrolebinding_seldon-manager-sas-rolebinding.yaml
19:  name: seldon-manager

helm-charts/seldon-core-operator/templates/clusterrolebinding_seldon-webhook-rolebinding.yaml
18:  name: seldon-manager
```

However, the Helm charts are generated which means there should be something upstream of them causing the hard-coded values; we need to change that too, else the next generation run of the charts will brush away these changes.

**Review by agrski — CHANGES_REQUESTED** [review]
As mentioned, we need to change that parts that are used to generate the Helm chart, else this change will soon be accidentally overridden.

**Comment by axsaucedo** [comment]
@BlaD200 will you get a chance to update?

**Comment by axsaucedo** [comment]
Closing for the time being but we can explore adding on the split resource file in the near future if you get a chance

**Pull request closed without merging**
